src/Components/Components/src/NavigationManager.cs

```diff
@@ -90,10 +90,47 @@ protected set
         /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
         /// (as returned by <see cref="BaseUri"/>).</param>
         /// <param name="forceLoad">If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.</param>
-        public void NavigateTo(string uri, bool forceLoad = false)
+        // 5.0 BACKCOMPAT OVERLOAD -- DO NOT TOUCH
+        public void NavigateTo(string uri, bool forceLoad)
         {
+            NavigateTo(uri, forceLoad, false);
+        }
+
+        /// <summary>
+        /// Navigates to the specified URI.
+        /// </summary>
+        /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
+        /// (as returned by <see cref="BaseUri"/>).</param>
+        /// <param name="forceLoad">If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.</param>
+        /// <param name="replace">If true, will replace the uri in the current browser history state, instead of pushing the new uri onto the browser history stack.</param>
+        public void NavigateTo(string uri, bool forceLoad = false, bool replace = false)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+            AssertInitialized();
+            NavigateToCore(uri, new NavigationOptions { ForceLoad = forceLoad, Replace = replace });
+        }
+
+        /// <summary>
+        /// Navigates to the specified URI.
+        /// </summary>
+        /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
+        /// (as returned by <see cref="BaseUri"/>).</param>
+        /// <param name="options">Add additional navigation options <see cref="NavigationOptions"/>.</param>
+        public void NavigateTo(string uri, NavigationOptions options)
```

Comment:
```suggestion
        public void NavigateTo(string uri, in NavigationOptions options)
```

Once it's a `readonly struct`, we can avoid copying by using `in`.